### PR TITLE
Add a spec file for pulp-puppet to Fedora 23.

### PR DESCRIPTION
**Pull request by bowlofeggs**

### CODE REVIEW

File: rpms/pulp-puppet/pulp-puppet.spec [hunk 1]
```diff
@@ -0,0 +1,245 @@
+%global required_pulp_version 2.8.2
+
+
+Name:		pulp-puppet
+Version:	2.8.2
+Release:	1%{?dist}
+
+License:	GPLv2
```
Comment (jeremycline) [inline rpms/pulp-puppet/pulp-puppet.spec]: GPLv2+

File: rpms/pulp-puppet/pulp-puppet.spec [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+install -pm644 pulp_puppet_handlers/etc/pulp/agent/conf.d/* \
+    %{buildroot}/%{_sysconfdir}/pulp/agent/conf.d/
+
+
+# ---- Admin Extensions
+%package admin-extensions
+Summary:	The Pulp Puppet admin client extensions
+Requires:	pulp-admin-client = %{required_pulp_version}
+Requires:	python2-pulp-puppet-common = %{version}
+Requires:	python2-setuptools
```
Comment (jeremycline) [inline rpms/pulp-puppet/pulp-puppet.spec:83]: Does this actually require setuptools? The only references I can find are in the setup.pys. This actually seems to apply to just about every package we ship.

File: rpms/pulp-puppet/pulp-puppet.spec [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+%dir %{python_sitelib}/pulp_puppet
+%{python_sitelib}/pulp_puppet/__init__.py*
+%{python_sitelib}/pulp_puppet/common/
+%dir %{python_sitelib}/pulp_puppet/extensions
+%{python_sitelib}/pulp_puppet/extensions/__init__.py*
+%{python_sitelib}/pulp_puppet_common*.egg-info
+
+
+%changelog
+* Mon May 09 2016 Randy Barlow <rbarlow@redhat.com> - 2.8.0-1
```
Comment (jeremycline) [inline rpms/pulp-puppet/pulp-puppet.spec]: Version doesn't match current version.

File: rpms/pulp-puppet/pulp-puppet.spec (hunk 1 above)
Comment (bowlofeggs) [inline rpms/pulp-puppet/pulp-puppet.spec]: Fixed, and in Fedora too.

File: rpms/pulp-puppet/pulp-puppet.spec (hunk 3 above)
Comment (bowlofeggs) [inline rpms/pulp-puppet/pulp-puppet.spec]: Fixed.

File: rpms/pulp-puppet/pulp-puppet.spec (hunk 2 above)
Comment (bowlofeggs) [inline rpms/pulp-puppet/pulp-puppet.spec:83]: The runtime Python package that is used to load our entry points is called pkg_resources, and it looks like that is provided by python2-setuptools (at least it is on Fedora Rawhide):

```
$ dnf provides /usr/lib/python2.7/site-packages/pkg_resources/__init__.py
python2-setuptools-20.9.0-1.fc25.noarch : Easily build and distribute Python packages
Repo        : rawhide
```

Comment (jeremycline) [inline rpms/pulp-puppet/pulp-puppet.spec:83]: Ahh, that makes sense.

Comment (bowlofeggs) [inline rpms/pulp-puppet/pulp-puppet.spec:83]: @jeremycline on second thought I think you are on to something though - only Pulp platform actually imports pkg_resources so pulp.server is the only package that should depend on it. All of the plugins are simply discovered by that, so they don't need the dependency as well. I will make this change everywhere.

Comment (bowlofeggs) [inline rpms/pulp-puppet/pulp-puppet.spec:83]: @jeremycline I created https://github.com/pulp/packaging/pull/21 to remove the unneeded requires lines from the plugins. I also fixed this in Fedora.

**Pull request merged**